src/atoms/Checkbox/Checkbox.native.js

```diff
@@ -0,0 +1,238 @@
+        default:
+          return 'small';
+      }
+    },
+  },
+  title: {
+    margin() {
+      return [0, 0, 0, 0.5];
+    },
+  },
+  backdrop: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '24px',
+            height: '24px',
+            borderRadius: '12px',
+          };
+        case 'medium':
+          return {
+            width: '20px',
+            height: '20px',
+            borderRadius: '10px',
+          };
+        case 'small':
+          return {
+            width: '16px',
+            height: '16px',
+            borderRadius: '8px',
+          };
+        case 'xsmall':
+          return {
+            width: '12px',
+            height: '12px',
+            borderRadius: '6px',
+          };
+        default:
+          return {
+            width: '20px',
+            height: '20px',
+            borderRadius: '10px',
+          };
+      }
+    },
+  },
+};
+
+const isPropDefined = (prop) => {
+  return typeof prop !== 'undefined';
+};
+
+const Checkbox = ({
+  defaultChecked,
+  checked,
+  onChange,
+  disabled,
+  title,
+  size,
+  variantColor,
+  helpText,
+  errorText,
+  testID,
+}) => {
+  let titleTextColor = 'shade.970';
+  let helpTextColor = 'shade.950';
+
+  let checkboxInitialState = false;
+
+  if (isPropDefined(defaultChecked)) {
+    checkboxInitialState = defaultChecked;
+  }
+
+  if (isPropDefined(checked)) {
+    checkboxInitialState = checked;
+  }
+
+  const [isChecked, setCheckboxState] = useState(checkboxInitialState);
+  const [underlayColor, setUnderlayColor] = useState('transparent');
+  const theme = useContext(ThemeContext);
+
+  const onPressIn = () => {
+    let colorKey = 'tone.940';
+    if (isChecked) {
+      colorKey = `${variantColor || 'primary'}.930`;
+    }
+    const newUnderlayColor = getColor(theme, colorKey);
+    setUnderlayColor(newUnderlayColor);
+  };
+
+  const onPressOut = () => {
+    setUnderlayColor('transparent');
+  };
+
+  const onPress = () => {
+    if (isPropDefined(checked)) {
+      onChange(!isChecked);
+      return;
+    }
+    setCheckboxState((prevCheckboxState) => {
+      onChange(!prevCheckboxState);
+      return !prevCheckboxState;
+    });
+  };
+
+  if (disabled) {
+    titleTextColor = 'shade.950';
+    helpTextColor = 'shade.930';
+  }
+
+  return (
+    <Flex alignSelf="flex-start">
+      <TouchableOpacity
+        activeOpacity={1}
+        accessibilityRole="checkbox"
+        onPress={onPress}
+        underlayColor="transparent"
+        disabled={disabled}
+        onPressIn={onPressIn}
+        onPressOut={onPressOut}
+        {...automation(testID)}
+      >
+        <Flex flexDirection="row" alignItems="center">
+          <View>
+            <Backdrop
+              width={styles.backdrop.dimensions(size).width}
+              height={styles.backdrop.dimensions(size).height}
+              borderRadius={styles.backdrop.dimensions(size).borderRadius}
```

Comment:
Why not this?

```suggestion
             {...styles.backdrop.dimensions(size)}
```

instead of calling same function three times?